runtime/sys.c

```diff
@@ -508,7 +508,17 @@ CAMLprim value caml_sys_executable_name(value unit)
   return caml_copy_string_of_os(caml_params->exe_name);
 }
 
-void caml_sys_init(const char_os * exe_name, char_os **argv)
+CAMLprim value caml_sys_proc_self_exe(value unit)
+{
+  if (caml_params->proc_self_exe == NULL)
+    return Val_none;
+  else
+    return caml_alloc_some(caml_copy_string_of_os(caml_params->proc_self_exe));
```

Comment:
@dra27 does this not to register the return of caml_copy_string_of_os as a root, as you are entering the GC twice in this function?

Reply:
The __function__ `caml_alloc_some` will register its argument as a root, so all is well, I think.

Reply:
Indeed, the root registration is done by `caml_alloc_some` (same as as the `caml_alloc_n` functions):
https://github.com/ocaml/ocaml/blob/27d26c9fdd885db8e1f34889207df3d1aa7db4bf/runtime/alloc.c#L281-L283